README.md

```diff
@@ -1,96 +1,200 @@
+
+When a notebook is executed, what is actually happening is that each code cell in
+the notebook is executed, in order, from top to bottom.
+
+Each code cell is selectable and is preceded by a tag in the left margin. The tag
+format is `In [x]:`. Depending on the state of the notebook, the `x` can be:
+
+* A blank, this indicates that the cell has never been executed.
+* A number, this number represents the relative order this code step was executed.
+* A `*`, this indicates that the cell is currently executing.
+
+There are several ways to execute the code cells in your notebook:
+
+* One cell at a time.
+  * Select the cell, and then press the `Play` button in the toolbar.
+* Batch mode, in sequential order.
+  * From the `Cell` menu bar, there are several options available. For example, you
+    can `Run All` cells in your notebook, or you can `Run All Below`, that will
+    start executing from the first cell under the currently selected cell, and then
+    continue executing all cells that follow.
+* At a scheduled time.
+  * Press the `Schedule` button located in the top right section of your notebook
+    panel. Here you can schedule your notebook to be executed once at some future
+    time, or repeatedly at your specified interval.
+
+## 6. Save and Share
+
+### How to save your work:
+
+Under the `File` menu, there are several ways to save your notebook:
+
+* `Save` will simply save the current state of your notebook, without any version
+  information.
+* `Save Version` will save your current state of your notebook with a version tag
+  that contains a date and time stamp. Up to 10 versions of your notebook can be
+  saved, each one retrievable by selecting the `Revert To Version` menu item.
+
+### How to share your work:
+
+You can share your notebook by selecting the `Share` button located in the top
+right section of your notebook panel. The end result of this action will be a URL
+link that will display a “read-only” version of your notebook. You have several
+options to specify exactly what you want shared from your notebook:
+
+* `Only text and output`: will remove all code cells from the notebook view.
+* `All content excluding sensitive code cells`:  will remove any code cells
+  that contain a *sensitive* tag. For example, `# @hidden_cell` is used to protect
+  your credentials from being shared.
+* `All content, including code`: displays the notebook as is.
+* A variety of `download as` options are also available in the menu.
+
+# Troubleshooting
+
+* Notebook error:
+
+  ![library-error](doc/source/images/library-error.png)
+  
+  This will occur if when you run the notebook multiple times. The custom library `NAME` found in the structure below must be unique for each run. Change the value and run the cell again.
+
+  ```bash
```

Comment:
use python instead here?

Reply:
will try that. Also noticed that the sentence needs some fixin